components/usage-api/typescript/BUILD.yaml

```diff
@@ -0,0 +1,29 @@
+packages:
+  - name: lib
+    type: yarn
+    deps:
+      - components/gitpod-protocol:lib
+    srcs:
+      - src/**
+      - package.json
+      - tsconfig.json
+    config:
+      packaging: library
+      dontTest: true
+      commands:
+        build: ["yarn", "build"]
+      yarnLock: ${coreYarnLockBase}/../yarn.lock
+      tsconfig: tsconfig.json
+
+  - name: publish
+    type: generic
+    env:
+      - DO_PUBLISH=${publishToNPM}
+    argdeps:
+      - npmPublishTrigger
+    deps:
+      - :lib
+      - components/gitpod-protocol:scripts
+    config:
+      commands:
+        - ["node", "components-gitpod-protocol--scripts/publish.js", "${version}", "components-usage-api-typescript--lib/package"]
```

Comment:
Where is this publish task used?

Reply:
disclaimer: I did a simple copy, paste and rename from some other `api` package.

We use that somewhere in the build process to be able to consume the NPM package during yarn builds, e.g. in `server`.